app/models/favourite.rb

```diff
@@ -0,0 +1,7 @@
+class Favourite < ActiveRecord::Base
+  validates :user_id, presence: true
+  validates :quote_id, presence: true, uniqueness: { scope: :user_id }
```

Comment:
For both of these, it's better to validate on the model instead of the column:

```
validates :user, presence: true
```
Which will check that the `user_id` is present *and* that it corresponds to a record in the `users` table. How you have it now, it just checks that *any* value is there.